### DISCUSSION TITLE
Weekly/Monthly Gas Cost Wrong

### DISCUSSION THREAD

**Discussion by Fainbrog** [BottlecapDave/HomeAssistant-OctopusEnergy#1107] [Q&A]
Hi all,

I think I'm seeing incorrect gas rates and thus costs in the weekly/monthly cost tracker.

Have set up gas_cost tracker, per the instructions, and when looking at the attributes, it's showing very high rate, eg;

Start: '2024-12-04T00:00:00+00:00' 
end: '2024-12-04T00:30:00+00:00' 
**rate: 0.1911** 
consumption: 0.72934
cost: 0.13937687399999998

So, rate is 19p where-as today is 6.22p or thereabouts.

Target entity is current consumption which shows the correct kWh usage compared to the Octopus app.

Current Accumulative Cost for gas is correct and electricity week/month is fine.

Have I set something up incorrectly? Have adjusted the calorific value, but the issue seems to be the cost rate.

**Comment by BottlecapDave** [comment]
Hello. The cost entity uses the tariffs that are configured against your account. The total value also includes the standing charge, but based on what you've posted, this isn't it. 

What did the current gas rate sensor state at the time? What tariff are you on?

**Reply by BottlecapDave** [reply]
Oh sorry I just re-read the issue. Cost trackers only support electricity meters, so it's using the electricity cost. I'll make this clearer in the docs. There isn't week/month sensors as people typically use the energy dashboard to track data greater than a day.

**Reply by Fainbrog** [reply]
Gotcha, thanks, I've not used the Energy Dashboard much, so, will have a look. 

I was hoping to be able to add an entity to my own dashboard with monthly cost and average rate/kWh which I've been able to do for electricity - just for the sake of nerding out on data as it's the first year on Tracker and Agile :)

I'll have to have a play with what data I can use/calculate in a template.

Thanks for the reply and for keeping up the development of the integration.

**Reply by Fainbrog** [reply]
I've recently revisited this and come up with a bit of a fudge, but, one that works well enough for me.

I've created a template/sensor that takes total kWh consumption from the gas monthly cost tracker and multiplies by the current rate gas cost. Ok, the latter fluctuates, but, it gives enough of a hint at the overall cost and allows me to put it all in one place on a dashboard for quick ref, rather than having to go to the Energy dashboard. As we are close to the end of the month, I just checked and it's about £3.50 off the actual figure, which when we are at c.£170 usage, I'm not going to argue with.

I also created estimated/predicted monthly costs for gas and electric, which are helpful for me. For gas, taking the total kWh consumption from the monthly cost tracker, multiplying by the current rate, divided by the day of the month (ie .27) and multiplied by the number of days in the month. For electric, taking accumulated electricity cost/day of the month (ie 27), multiplied by number of days in the month.

Clearly, I have too much time on my hands and too much of a propensity to tinker with HA..
